dota2/features/lobby.py

```diff
@@ -0,0 +1,221 @@
+from dota2.enums import ESOType, EGCBaseMsg, EDOTAGCMsg
+
+
+class Lobby(object):
+    EVENT_LOBBY_INVITE = 'lobby_invite'
+    """When a lobby invite is received
+    :param message: `CSDOTALobbyInvite <https://github.com/ValvePython/dota2/blob/ca75440adca20d852b9aec3917e4387466848d5b/protobufs/dota_gcmessages_common_match_management.proto#L100>`_
+    :type message: proto message
+    """
+    EVENT_LOBBY_INVITE_END = 'lobby_invite_end'
+    """When a lobby invite is no longer valid
+    :param message: `CSDOTALobbyInvite <https://github.com/ValvePython/dota2/blob/ca75440adca20d852b9aec3917e4387466848d5b/protobufs/dota_gcmessages_common_match_management.proto#L100>`_
+    :type message: proto message
+    """
+    EVENT_LOBBY_NEW = 'lobby_new'
+    """Entered a lobby, either by creating one or accepting an invite
+
+    :param message: `CSODOTALobby <https://github.com/ValvePython/dota2/blob/ca75440adca20d852b9aec3917e4387466848d5b/protobufs/dota_gcmessages_common_match_management.proto#L193>`_
+    :type message: proto message
+    """
+    EVENT_LOBBY_CHANGED = 'lobby_changed'
+    """Anything changes to the lobby state, players, options, broadcasters...
+
+    :param message: `CSODOTALobby <https://github.com/ValvePython/dota2/blob/ca75440adca20d852b9aec3917e4387466848d5b/protobufs/dota_gcmessages_common_match_management.proto#L193>`_
+    :type message: proto message
+    """
+
+    lobby = None
+    lobby_invite = {}
```

Comment:
SO stands for shared object, which is just a shared state. SOCache is essentially a `dict` and deals with creating/updating/removing them. For us, they are read only. We can only change state indirectly by asking GC to do something.